docusaurus/docs/Android/04-compose/02-general-customization/02-attachment-factory.mdx

```diff
@@ -3,42 +3,29 @@
-    private val predicate: (Message) -> Boolean,
-) {
-    // Validator to check if we can handle the given message
-    public fun canHandle(message: Message): Boolean {
-        return predicate(message)
-    }
-}
+public abstract class AttachmentFactory(
+    public val canHandle: (attachments: List<Attachment>) -> Boolean,
+    public val content: @Composable (AttachmentState) -> Unit,
+)
 ```
 
-* `factory`: Defines the composable function that accepts an `AttachmentState` and shows any given attachment component.
-* `predicate`: Lambda function that accepts a message and returns `true` if a given factory can consume the message to show an attachment. The first factory in the list of available factories that can handle the message will be used to render its attachments.
+* `canHandle`: Lambda function that accepts a list of attachments in a message and returns `true` if a given factory can consume the attachments and show some UI.
+* `content`: Defines the composable function that accepts an `AttachmentState` and shows a given attachment component.
 
-There are three examples of default attachment factory implementations, in the [`StreamAttachmentFactories.kt` file](https://github.com/GetStream/stream-chat-android/blob/main/stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/theme/StreamAttachmentFactories.kt):
+There are a few examples of default attachment factory implementations, in the [`StreamAttachmentFactories.kt` file](https://github.com/GetStream/stream-chat-android/blob/main/stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/theme/StreamAttachmentFactories.kt):
 
 ```kotlin
 public val defaultFactories: List<AttachmentFactory> = listOf(
-    AttachmentFactory(
-        { state -> LinkAttachmentFactory(state) },
-        { message -> message.attachments.any { it.ogUrl != null } }
-    ),
-    AttachmentFactory(
-        { state -> ImageAttachmentFactory(state) },
-        { message -> message.attachments.all { it.type == "image" } }
-    ),
-    AttachmentFactory(
-        { state -> FileAttachmentFactory(state) },
-        { message -> message.attachments.any { it.type != "image" } }
-    )
+    LinkAttachmentFactory(),
+    GiphyAttachmentFactory(),
+    ImageAttachmentFactory(),
+    FileAttachmentFactory()
 )
 ```
 
-These factories perform specific checks, like if the `ogUrl` exists, giving us information that this is a link attachment, or if all of the message attachments are of the `"image"` type.
+These factories perform specific checks that you can explore by opening each specific factory.
 
-Each of these factories supplies a validator, as well as a factory composable lambda function that calls `LinkAttachmentFactory`, `ImageAttachmentFactory` or `FileAttachmentFactory`.
+Each of these factories supplies a predicate, as well as a content composable lambda function that calls `LinkAttachmentContent`, `ImageAttachmentContent` or `FileAttachmentContent`.
```

Comment:
We don't list `GiphyAttachmentFactory` here, but maybe this is a lesson in how we shouldn't try listing things like this in docs 😀

Reply:
True, I've gone through the docs around to see if I can update it, but I've missed this point, I'll update accordingly.